Makefile

```diff
@@ -12,7 +12,7 @@ src/Lex.hs: src/Lex.x
 
 # builds the rest of the haskell files (compiler)
 bin/act: src/*.hs
-	cd src && cabal new-update && cabal v2-install --installdir=../bin --overwrite-policy=always && cd ..
```

Comment:
The `new-update` is an unfortunate workaround to a confusing purity leak in the `nix-shell`, CI currently fails without it. I think it's nice to remove it from the `Makefile` because it's very annoying to wait for the update, but maybe lets add it to the github workflow instead?

Reply:
https://github.com/ethereum/act/pull/44/commits/1d765b1402e7e21b0fdd8efba17a155eb6f70166 should be sufficient, right?

Reply:
yes :pray: :sparkles: